modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/XmlExtensionLoaderDelegate.java

```diff
@@ -475,4 +487,30 @@ private MetadataType extractType(String receivedType) {
     }
     return metadataType.get();
   }
+
+  private void declareErrorModels(OperationDeclarer operationDeclarer, XmlDslModel xmlDslModel, String operationName,
+                                  ComponentModel operationModel) {
+    Optional<ComponentModel> optionalParametersComponentModel = operationModel.getInnerComponents()
+        .stream()
+        .filter(child -> child.getIdentifier().equals(OPERATION_ERRORS_IDENTIFIER)).findAny();
+    optionalParametersComponentModel.ifPresent(componentModel -> componentModel.getInnerComponents()
+        .stream()
+        .filter(child -> child.getIdentifier().equals(OPERATION_ERROR_IDENTIFIER))
+        .forEach(param -> {
+          final String namespace = xmlDslModel.getPrefix().toUpperCase();
+          final String typeName = param.getParameters().get(ERROR_TYPE_ATTRIBUTE);
+          if (StringUtils.isBlank(typeName)) {
+            throw new IllegalModelDefinitionException(format("The operation [%s] cannot have an <error> with an empty 'type' attribute",
+                                                             operationName));
+          }
+          if (typeName.contains(NAMESPACE_SEPARATOR)) {
+            throw new IllegalModelDefinitionException(format("The operation [%s] cannot have an <error> [%s] that contains a reserved character [%s]",
+                                                             operationName, typeName,
+                                                             NAMESPACE_SEPARATOR));
+          }
+          operationDeclarer.withErrorModel(ErrorModelBuilder.newError(typeName, namespace)
+              .withParent(ErrorModelBuilder.newError(Errors.ComponentIdentifiers.ANY).build())
```

Comment:
it wont. 
It's messy to use the declarative part of a connector to define a hierarchy (discussed already with PLG)

Both smart connectors and runtime should have a place to declare the errors, and their hierarchy, and then use them as references.